### PR TITLE
macros/os_switch: consider not only dirs but also files, add README

### PR DESCRIPTION
**Pull request by jan--f**
Signed-off-by: Jan Fajerski <jfajerski@suse.com>

This allows to either add, say a SUSE.sls or a more complex SUSE directory with several more complex sls files.

### CODE REVIEW

File: srv/salt/ceph/macros/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+include:
+  - .{{ os_switch(custom) }}
+```
+
+Use this macro to implement an opportunistic operating system switch in your
+state files. The macro will check the directory of the including state files and
+depending on the content output a single state file or directory to include. The
+rules are as follows:
+- Files are preferred directories. However the directory include can be forced
```
Comment (alexandergraul) [inline srv/salt/ceph/macros/README.md]: You forgot the "over" in:
> Files are preferred __over__ directories

File: srv/salt/ceph/macros/os_switch.sls [hunk 2]
```diff
@@ -5,12 +5,20 @@
   {%- set osrelease = os + "-" + grains.get('osrelease') -%}
   {%- if salt['file.directory_exists'](abspath + "/" +  custom) -%}
     {{ custom }}
+  {%- elif salt['file.file_exists'](abspath + "/" +  custom + ".sls") -%}
```
Comment (jschmid1) [inline srv/salt/ceph/macros/os_switch.sls]: I'm aware that a 'path' should not end with a '/' but in this case we are not using it as a standalone variable which makes all if-clauses a bit redundant:

`abspath + "/" +  custom + ".sls"`

vs
` {% set abspath = "/srv/salt/" + slspath + '/' %}`

and then 

`abspath +  custom + ".sls"`

not sure if I like it either :)

File: srv/salt/ceph/macros/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  by explicitly including the directory's init.sls
+- The following names are tested in order and returned if either file or
+  directory is present in the including sls file's directory.
+- ```custom```
```
Comment (jschmid1) [inline srv/salt/ceph/macros/README.md]: nitpicky:

maybe use extra indentation here? Technically it's a enumeration within a enumeration:

- The following names are tested in order and returned if either file or￼ directory is present in the including sls file's directory.
  - value of the ```osfinger``` grain, for example ```CentOS```
  - etc..

File: srv/salt/ceph/macros/os_switch.sls (hunk 2 above)
Comment (jan--f) [reply srv/salt/ceph/macros/os_switch.sls]: :+1:

File: srv/salt/ceph/macros/README.md (hunk 1 above)
Comment (jan--f) [reply srv/salt/ceph/macros/README.md]: thx!

**Pull request merged**